PWG/EMCAL/EMCALtasks/AliEmcalCorrectionClusterNonLinearityMCAfterburner.cxx

```diff
@@ -0,0 +1,552 @@
+// AliEmcalCorrectionClusterNonLinearityMCAfterburner.cxx
+//
+
+#include "AliEmcalCorrectionClusterNonLinearityMCAfterburner.h"
+#include <TList.h>
+#include "AliClusterContainer.h"
+#include "AliFiducialCut.h"
+
+/// \cond CLASSIMP
+ClassImp(AliEmcalCorrectionClusterNonLinearityMCAfterburner);
+/// \endcond
+
+// Actually registers the class with the base class
+RegisterCorrectionComponent<AliEmcalCorrectionClusterNonLinearityMCAfterburner> AliEmcalCorrectionClusterNonLinearityMCAfterburner::reg("AliEmcalCorrectionClusterNonLinearityMCAfterburner");
+
+/**
+ * Default constructor
+ */
+AliEmcalCorrectionClusterNonLinearityMCAfterburner::AliEmcalCorrectionClusterNonLinearityMCAfterburner() :
+  AliEmcalCorrectionComponent("AliEmcalCorrectionClusterNonLinearityMCAfterburner"),
+  fEnergyDistBefore(0),
+  fEnergyTimeHistBefore(0),
+  fEnergyDistAfter(0),
+  fEnergyTimeHistAfter(0),
+  fAfterburnerVersion(0),
+  fMCPeriod(""),
+  fSetForceClusterE(kFALSE)
+{
+}
+
+/**
+ * Destructor
+ */
+AliEmcalCorrectionClusterNonLinearityMCAfterburner::~AliEmcalCorrectionClusterNonLinearityMCAfterburner()
+{
+}
+
+/**
+ * Initialize and configure the component.
+ */
+Bool_t AliEmcalCorrectionClusterNonLinearityMCAfterburner::Initialize()
+{
+  // Initialization
+  AliEmcalCorrectionComponent::Initialize();
+
+  AliWarning("Init EMCAL non-linearity MC afterburner");
+
+  GetProperty("afterburnerVersion", fAfterburnerVersion); //Fixme there are only 4 cases, should be fixed probably
+  GetProperty("setMCPeriod", fMCPeriod);
```

Comment:
It's not a perfect idea, but in principle we could map with run numbers, since we have those at hand automatically. It seems to me that if we implement the period directly, it should be in the base component...

Reply:
Yeah, that's an option, but I don't think it will be so straightforward for two reasons:

- we then have to maintain the run number to period map, which isn't great.
- there are multiple MC productions anchored to a single data period. So somehow we would need to find a way to invert a many-to-one map. I don't know how we would do that

Reply:
There were different passes in the original class but I deleted this from my copying, since we should anyway only use the latest pass. I also have to say that they (the creators of this method) have on their ToDo list to evaluate whether they need different parameters for different periods at all. But until we know for sure this was their way of handling it.